tests/e2e/default/sms/rapidpro.wdio-spec.js

```diff
@@ -0,0 +1,593 @@
+      } catch (err) {
+        expect(err.responseBody).to.eql({ code: 403, error: 'Missing authorization token' });
+      }
+    });
+
+    it('should fail with incorrect authentication', async () => {
+      await setIncomingKey();
+      await utils.updateSettings({ sms: smsSettings }, true);
+
+      try {
+        await utils.request({
+          path: endpoint,
+          method: 'POST',
+          headers: { authorization: 'Token not the correct key' },
+          noAuth: true,
+        });
+        throw new Error('should have thrown');
+      } catch (err) {
+        expect(err.responseBody).to.eql({ code: 403, error: 'Incorrect token' });
+      }
+    });
+
+    it('should fail with malformed authentication', async () => {
+      await setIncomingKey();
+      await utils.updateSettings({ sms: smsSettings }, true);
+
+      try {
+        await utils.request({
+          path: endpoint,
+          method: 'POST',
+          headers: { authorization: `Tokens ${INCOMING_KEY}` },
+          noAuth: true,
+        });
+        throw new Error('should have thrown');
+      } catch (err) {
+        expect(err.responseBody).to.eql({ code: 403, error: 'Missing authorization token' });
+      }
+    });
+
+    it('should fail when message was not created', async () => {
+      await setIncomingKey();
+      await utils.updateSettings({ sms: smsSettings }, true);
+
+      try {
+        await utils.request({
+          path: endpoint,
+          method: 'POST',
+          headers: { authorization: `Token ${INCOMING_KEY}` },
+          noAuth: true,
+          body: { from: 'phone', content: 'test content' },
+        });
+        throw new Error('should have thrown');
+      } catch (err) {
+        expect(err.responseBody).to.eql({ code: 400, error: 'Message was not saved' });
+      }
+    });
+
+    it('should create messages', async () => {
+      const smsId = 'the_gateway_ref';
+      const rawPhone = '+50689999999';
+      const smsContent = 'the sms message content';
+
+      const message = {
+        id: smsId,
+        from: rawPhone,
+        content: smsContent,
+      };
+
+      await setIncomingKey();
+      await utils.updateSettings({ sms: smsSettings }, true);
+      const messageResult = await utils.request({
+        path: endpoint,
+        method: 'POST',
+        headers: { authorization: `Token ${INCOMING_KEY}` },
+        noAuth: true,
+        body: message,
+      });
+
+      expect(messageResult).to.eql({ saved: 1 });
+
+      await loginPage.cookieLogin();
+      await commonPage.goToMessages();
+
+      const { heading, summary} = await messagesPage.getMessageInListDetails(rawPhone);
+      expect(heading).to.equal(rawPhone);
+      expect(summary).to.equal(smsContent);
+
+      await messagesPage.openMessage(rawPhone);
+      const { name } = await messagesPage.getMessageHeader();
+      expect(name).to.equal(rawPhone);
+
+      const { content, state, dataId } = await messagesPage.getMessageContent(1);
```

Comment:
Nice job working the `dataId` into the existing utils instead of adding more custom code for it!  :star_struck: